sdk/test-utils/recorder/src/baseRecorder.ts

```diff
@@ -0,0 +1,389 @@
+  if (!isBrowser() && (isRecording || isPlayingBack)) {
+    nock = require("nock");
+  }
+
+  if (isBrowser() && isRecording()) {
+    customConsoleLog();
+  }
+
+  if (isPlayingBack()) {
+    // Providing dummy values to avoid the error [ENVs for storage packages]
+    env.ACCOUNT_NAME = "fakestorageaccount";
+    env.ACCOUNT_KEY = "aaaaa";
+    env.ACCOUNT_SAS = "aaaaa";
+    env.STORAGE_CONNECTION_STRING = `DefaultEndpointsProtocol=https;AccountName=${
+      env.ACCOUNT_NAME
+    };AccountKey=${env.ACCOUNT_KEY};EndpointSuffix=core.windows.net`;
+  }
+}
+
+const skip = [
+  // Abort
+  "browsers/aborter/recording_should_abort_after_aborter_timeout.json"
+];
+
+export abstract class BaseRecorder {
+  protected readonly filepath: string;
+  public uniqueTestInfo: TestInfo = { uniqueName: {}, newDate: {} };
+
+  constructor(runtime: string, testHierarchy: string, testTitle: string, ext: string) {
+    this.filepath =
+      runtime +
+      "/" +
+      this.formatPath(testHierarchy) +
+      "/recording_" +
+      this.formatPath(testTitle) +
+      "." +
+      ext;
+  }
+
+  protected formatPath(path: string): string {
+    return path
+      .toLowerCase()
+      .replace(/ /g, "_")
+      .replace(/<=/g, "lte")
+      .replace(/>=/g, "gte")
+      .replace(/</g, "lt")
+      .replace(/>/g, "gt")
+      .replace(/=/g, "eq")
+      .replace(/\W/g, "");
+  }
+
+  /**
+   * Additional layer of security to avoid unintended/accidental occurrences of secrets in the recordings
+   * */
+  protected filterSecrets(recording: string): string {
+    let updatedRecording = recording;
+    for (const k of Object.keys(replaceableVariables)) {
+      const escaped = escapeRegExp(env[k]);
+      updatedRecording = updatedRecording.replace(
+        new RegExp(escaped, "g"),
+        replaceableVariables[k]
+      );
+    }
+    for (const map of replacements) {
+      updatedRecording = map(updatedRecording);
+    }
+
+    // Handling storage environment variables separately
+    updatedRecording = recording.replace(new RegExp(env.ACCOUNT_NAME, "g"), "fakestorageaccount");
+    if (env.ACCOUNT_KEY) {
+      updatedRecording = updatedRecording.replace(new RegExp(env.ACCOUNT_KEY, "g"), "aaaaa");
+    }
+    if (env.ACCOUNT_SAS) {
+      updatedRecording = updatedRecording.replace(
+        new RegExp(env.ACCOUNT_SAS.match("(.*)&sig=(.*)")[2], "g"),
+        "aaaaa"
+      );
+    }
+    return updatedRecording;
+  }
+
+  public skip(): boolean {
+    return skip.includes(this.filepath);
+  }
+
+  public abstract record(): void;
+  public abstract playback(filePath: string): void;
+  public abstract stop(): void;
+}
+
+export class NockRecorder extends BaseRecorder {
+  constructor(testHierarchy: string, testTitle: string) {
+    super("node", testHierarchy, testTitle, "js");
+  }
+
+  public record(): void {
+    nock.recorder.rec({
+      dont_print: true
+    });
+  }
+
+  public playback(filePath: string): void {
+    let path = require("path");
```

Comment:
Add a comment as to why we need to dynamically import this

Reply:
Added